### ISSUE TITLE
Use the metrics to calculate carbon footprint and calories burned

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-phone#80]
For calories burned, we need to use values from the "Compendium of Physical Activities".
https://sites.google.com/site/compendiumofphysicalactivities/corrected-mets

We can calculate a default that is not personalized just based on the speed, and allow users to personalize it based on age, body mass, and sex. The raw METs and the personalization formula are available at the link above.

Recall that with d3, you can build the visualization using icons. Crazy idea - how about converting it to food instead? i.e. you biked so much this week that you can have an extra slice of chocolate cake, or something. 
http://well.blogs.nytimes.com/2016/07/18/an-app-to-deconstruct-your-food

For the carbon footprint calculation, we can just re-use what I did last time.
https://github.com/e-mission/e-mission-server/blob/master/front/server/carbon_calc_details.html
https://github.com/e-mission/e-mission-server/blob/master/emission/analysis/result/carbon.py

The range will be assuming (all IN_VEHICLE = train <-> all IN_VEHICLE = car)

Similar to the calorie counter, you can allow people to enter their make and model of car to compute their MPG.
http://www.fueleconomy.gov/feg/ws/index.shtml
which you can convert to GHG and use instead of the value for car.

The conversion is already in carbon.py.

**Comment by shankari** [comment]
@krisma this is all yours

**Comment by shankari** [comment]
wrt crazy idea, public source of information for calories in various food items is:
https://ndb.nal.usda.gov/ndb/doc/index
so you could ask the user to pick a favorite food equivalent instead of calories (like ice cream)
then look up the calories for ice cream (say):
https://ndb.nal.usda.gov/ndb/foods/show/202?manu=&fgcd=

and then show the calories in ice cream equivalents.

**Comment by shankari** [comment]
@krisma if you are doing the food equivalent, please let me know so that I can sign up for the API key. I want the API key to be registered to me for continuity.

**Comment by shankari** [comment]
also, @juemura, have you seen the "food equivalent" concept before?
do you want to include it in the study (how many people choose the "food equivalent"? how much does it help?)

**Comment by juemura** [comment]
I have, though never thought much of it. Do you think this will be ready by Friday? If so, then yes. I can include a question in the pre-survey about whether they care to see calories/food equivalent info, or we can just leave it and ask about it at the end.

**Comment by shankari** [comment]
wrt the game, it seems like the idea of the "food equivalent" could also be thought of as a reward that you get for calories. Maybe you can earn an slice of cake by biking instead of driving?

The problem with that kind of reward based mechanism is that people tend to underestimate the calories eaten and overestimate the calories burned, so they actually end up gaining weight. So dunno if we want to go down that path.

What do people think?

**Comment by shankari** [comment]
Also, couple of other brain-dumped comments.

There are existing real-world goals that we can leverage. I am not sure whether we should visualize those goals or incorporate them into the game (or both).

I had some of these goals in the old carbon footprint calculator. They are also described in:
https://github.com/e-mission/e-mission-server/blob/master/front/server/carbon_calc_details.html

- optimal: all short trips (< 3km) via walk/bike, all long trips via train
- all drive: self explanatory
- SB 375 goal: per capita VMT reduction goals through land use and urban planning from SB 375
http://www.arb.ca.gov/cc/sb375/sb375.htm
- CA 2050 goal: Estimate of 80% below 1990 goals, calculated by me.

Also for exercise, there are CDC recommendations (roughly 150 mins of walking or 75 mins of running). We might be able to find better metrics with speeds, or that correlate to calories if we look further.
http://www.cdc.gov/physicalactivity/basics/index.htm

We should also plan on adding descriptions for how some of the metrics were calculated - e.g. calories burned, GHG, goals, etc.

One of the big differences between the commercial apps and us is that we have no secrets. You want to know how we calculate the metric - here it is.

And if you think that the calculation is wrong, and you have a better idea of how to do it, send us a pull request!!

**Comment by shankari** [comment]
The good thing about these goals is that they have some real world meanings, they are not just make believe. So for example, as a civilization, we really do need to get our GHG emissions down by 80% by 2050. The laws of physics are not amenable to argument :)

I am not sure exactly how to conceptualize that though.

**Comment by juemura** [comment]
You may be on to something big here. This reminds me of that conversation
with David, and I think this is something that has a lot of potential. If
we can provide this type of information with data and tools for people to
learn more, engage, set personal and community goals, etc, that itself is a
breakthrough if we can make it truly accessible and useful. It may be
enough that we raise awareness and make it easy for individuals to take
steps to participate in creating more sustainable transportation, but most
likely that would be just the first step to the real breakthrough, which is
to create a platform for individuals to start planning as a community
things that have been historically thought of as individual.
If you do get a class in the Fall, it may be worth spending a good amount
of time exploring effective ways to create this type of communication and
learning rather than focusing on engagement per se.
Anyway, just random thoughts.

**Comment by shankari** [comment]
for carbon footprint
--------------------
must have:
- range of carbon footprint based on standard values for train and car
- comparison with external goals - optimal, all drive, CA 2035 goal, CA 2050 goal

nice to have:
- allow you to specify your car and use the related MPG for calculation

**Comment by shankari** [comment]
MPG -> carbon footprint calculation is at:
https://github.com/e-mission/e-mission-server/blob/9e5717e75d03c817f78c9699fd143416bfe67215/emission/core/wrapper/user.py#L8

**Comment by shankari** [comment]
If user does enter MPG, you could also add a new comparison against driving an SUV.
So there would be "all drive" and "all drive (SUV)"

**Comment by shankari** [comment]
you could also add at the lower end, the most fuel efficient gasoline powered car.
Smart For two?
"all drive (super compact)"

**Comment by shankari** [comment]
EPA actually has a best and worst mileage estimator
http://www.fueleconomy.gov/feg/best-worst.shtml

And it does have electric cars - it provides MPGe (MPG-equivalent) for them. Lots of the efficient cars are electric, as one might expect (not just because of the the no tailpipe emissions, but also because of the more efficient drivetrain).

**Comment by shankari** [comment]
@krisma, note that the Tesla is not the most efficient electric vehicle - the Leaf is more efficient than it :)

**Comment by shankari** [comment]
Initial implementation of calorie counter and carbon footprint calculator were in https://github.com/e-mission/e-mission-phone/pull/88. Leaving this open in case it is useful for #107, specially the carbon footprint calculation.

**Comment by shankari** [comment]
We should probably close after #107 is implemented.

**Comment by fedpettinella** [comment]
I think a part of this solution will be to allow users to categorise "incidents"* by method of transport. The average speed is a very good indicator but it doesn't seem to be implemented yet, at least not on the Android app that I'm using. And the app might confuse bus, train, and car if it were to use average speed so giving the user editing permissions for transport type would result in a higher degree of precision.

* Or whatever you're calling a single start/stop movement of some kind. I think they're incidents.

**Comment by shankari** [comment]
> Or whatever you're calling a single start/stop movement of some kind. I think they're incidents.

No, they are trips. incidents are things that happen along a trip (e.g. people cutting you off, badly designed infrastructure, bumpy sidewalks, etc)

**Comment by shankari** [comment]
> The average speed is a very good indicator but it doesn't seem to be implemented yet, at least not on the Android app that I'm using. And the app might confuse bus, train, and car if it were to use average speed so giving the user editing permissions for transport type would result in a higher degree of precision.

This is part of the `cci` branch (https://github.com/e-mission/e-mission-phone/pull/250, https://github.com/e-mission/e-mission-phone/pull/256), but they have not yet been merged to master.

**Comment by shankari** [comment]
>  but it doesn't seem to be implemented yet, at least not on the Android app that I'm using

There is some minimal automatic sensing as part of the post-processing, but it does not distinguish between motorized modes :)
